packages/grafana-data/src/utils/dataLinks.ts

```diff
@@ -67,11 +67,13 @@ export function mapInternalLinkToExplore(options: LinkToExploreOptions): LinkMod
  */
 function generateInternalHref<T extends DataQuery = any>(datasourceName: string, query: T, range: TimeRange): string {
   return locationUtil.assureBaseUrl(
-    `/explore?left=${serializeStateToUrlParam({
-      range: range.raw,
-      datasource: datasourceName,
-      queries: [query],
-    })}`
+    `/explore?left=${encodeURIComponent(
```

Comment:
RichHistory also creates a URL with `serializeStateToUrlParam` [here](https://github.com/grafana/grafana/blob/main/public/app/core/utils/richHistory.ts#L184). Would it make sense to encode it too?

Reply:
Seems like this utils actually does that already: 
https://github.com/grafana/grafana/blob/main/public/app/core/utils/richHistory.ts#L186

It uses some custom one that seem to be concerned with angular js way it is done. Probably not needed for the Explore url though.